force-app/main/default/classes/ALLO_AllocationsRecalculateService_TEST.cls

```diff
@@ -620,6 +620,109 @@ public class ALLO_AllocationsRecalculateService_TEST {
         ALLO_UnitTestHelper_TEST.assertPaymentAllocation(pmtAlloc1, pmts[0].Id, 50.01, percentGAU2, gaus[1].Id, 'Payment Allocation For GAU[1]');
     }
 
+     /**
+     * @description This is a copy of confirmRecalculateServicePreservesFullPercentageWhenDefaultingPaymentAllocationsFromParent
```

Comment:
It might be worth eliminating the "This is a copy" with confirms that Allocations are no longer defaulted on Payments when sourceAmount = 0 so that Divide by Zero error no longer occurs.

Reply:
Done